File: python/0452用最少数量的箭引爆气球.py

points = [[1,2],[3,4],[5,6],[7,8]]

def findMinArrowShots(points: list) -> int:
    points=sorted(points,key=lambda x:x[0])
    print(points)
    res=1
    r=points[0][1]
    
    for p in points[1:]:
        if(p[0]>r):
            res+=1
            r=p[1]
        elif(p[1]<r):
            r=p[1]

    return res


print(findMinArrowShots((points)))

File: cpp/0452用最少数量的箭引爆气球.cpp
#include<vector>
#include<algorithm>

using namespace std;

static bool Cmp(const vector<int> &v1, const vector<int> &v2)
{
    return v1[0] < v2[0];
}

    int findMinArrowShots(vector<vector<int>> &points)
{

    sort(points.begin(), points.end(), Cmp);
    int res = 1;
    int r = points[0][1];

    for (int i = 1; i < points.size(); i++)
    {
        if(points[i][0]>r)
        {
            res++;
            r = points[i][1];
        }
        else if (points[i][1]<r)
        {
            r = points[i][1];
        }
    }

    return res;
}